app/components/layout/UserItem.tsx

```diff
@@ -45,16 +47,55 @@ export function UserItem() {
         </div>
       </Popover>
     );
+  }
+
+  const authError = searchParams.get("authError");
+  const closeAuthErrorDialog = () => {
+    searchParams.delete("authError");
+    setSearchParams(searchParams);
+  };
 
   return (
-    <form action={LOG_IN_URL} method="post" data-cy="log-in-form">
-      <button
-        type="submit"
-        className="layout__log-in-button"
-        data-cy="log-in-button"
-      >
-        <DiscordIcon /> {t("header.login")}
-      </button>
-    </form>
+    <>
+      <form action={LOG_IN_URL} method="post" data-cy="log-in-form">
+        <button
+          type="submit"
+          className="layout__log-in-button"
+          data-cy="log-in-button"
+        >
+          <DiscordIcon /> {t("header.login")}
+        </button>
+      </form>
+      {authError != null && (
```

Comment:
Oh, I get that, but this is actually the only case where I purposefully do write `==`/`!=`, because `== null` is specifically true for undefined and null, but nothing else.

So in my code style I usually use `== null` as a shorthand for just checking "is this null or undefined", without having to care about finding out which specific variant is relevant in this case, nor having to deal with special cases for 0 or "".

It's basically the same check the nullish coalescing operator (`??`) uses

But all that aside, I can definitely change this to be more specific, I just wanted to explain where I'm coming from :)